### ISSUE TITLE
Missing ? energy_sites/grid_import_export

### ISSUE THREAD

**Issue by avecezar** [jasonacox/pypowerwall#69]
Is it possible to add the following to READ and SET 

https://owner-api.teslamotors.com/api/1/energy_sites/{}/grid_import_export
Payload options:
"customer_preferred_export_rule": "battery_ok"
"customer_preferred_export_rule": "pv_only"
"disallow_charge_from_grid_with_solar_installed": false
"disallow_charge_from_grid_with_solar_installed": true

**Comment by jasonacox** [comment]
Hi, @avecezar , I assume you are talking about the [tools](https://github.com/jasonacox/pypowerwall/tree/main/tools) - Yes, these could be added. I often thought it would be good to have a single command line tool to do all these updates or polls. I'll flag that as a TODO and add these.  

If anyone wants to take this on and combine/expand any of the `set-mode` or `set-reserve` tools, please feel free and submit a PR. I won't have time to work on it for a while.

**Comment by avecezar** [comment]
Dear Jason, 

The “tools” concept and execution worked for me 100% (I am new to python and linux). 
What I would like to achieve is to stop charging at 70% if I set it this way using cron . 
So battery is charging from 23:30 cron is checking it every 10min and when the battery reached 70% it will disable "grid charging" making space for solar. At 6AM it would enable “grid charging” back again

Do you know any other way to do it ?

Cezar

**Comment by jasonacox** [comment]
Yes, that would be a new tool (to turn on/off grid charging). I suspect much of the code is already in one of the existing tools, but someone needs to investigate how to poll and change the "grid charing" setting.

**Comment by jasonacox-sam** [comment]
## Closing — Partially Addressed / Stale

**Issue summary:** Request to add support for the Tesla `energy_sites/{id}/grid_import_export` cloud API endpoint, specifically to read and set `customer_preferred_export_rule` (battery_ok / pv_only) and `disallow_charge_from_grid_with_solar_installed` (true/false).

**What happened:** Jason acknowledged the request and flagged it as a TODO. The use case was scheduling grid charging on/off via cron (e.g., charge to 70% overnight, then disable grid charging to make room for solar). Since this issue was opened, the pypowerwall library has gained built-in support for these operations:

- `pw.get_grid_charging()` / `pw.set_grid_charging(on)` — control whether Powerwall charges from grid
- `pw.get_grid_export()` / `pw.set_grid_export(on)` — control grid export behavior
- Proxy endpoints: `POST /control/grid_charging` and `POST /control/grid_export`
- Available in cloud, FleetAPI, and TEDAPI/v1r modes

The original request for a dedicated CLI tool wrapping these wasn't implemented, but the underlying API support is now in the library. Anyone needing cron-based control can script against these methods.

**Still have questions?** Feel free to reopen or start a new discussion. Any updates that help the community are always welcome.

---
*— Sam 🌟 (pypowerwall agent)*
